bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/StreamValue.java

```diff
@@ -121,14 +125,22 @@ public void publish(RefValue data) {
      * @param functionPointer represents the function pointer reference for the function to be invoked on receiving
      *                        messages
      */
-    public void subscribe(Consumer functionPointer) {
+    public void subscribe(FPValue<Object[], Object> functionPointer) {
+        BType[] parameters = ((BFunctionType) functionPointer.type).paramTypes;
+        int lastArrayIndex = parameters.length - 1;
+        if (!TypeChecker.isAssignable(constraintType, ((BFunctionType) functionPointer.getType())
```

Comment:
IMO we can replace this by checkIsLikeType method. So we do not need isAssignable .

Reply:
We also need to update those methods with stream type.

Reply:
@warunalakshitha CheckIsType is used to check if the given value can be assigned to a given type.. In this method we check two types.

Reply:
Removed redundant code